### ISSUE TITLE
As PO I want to streamline all environment variables.

### ISSUE THREAD

**Issue by joeribekker** [open-zaak/open-zaak#1629]
### Thema / Theme

Other

### Omschrijving / Description

The end goal is to streamline the environment variables but first task is to create a comparison matrix.

* [x] Please make a matrix that includes all environment variables of Open Zaak, Open Notificaties, Open Klant, Objects and Objecttypes API. This document can be an internal one, or we can make it part of Open API framework.
* [ ] Bring other settings files in line with eachother
* [ ] #1649

After this, we can compare and decide how to proceed.

PRs:
- [x] Open Klant https://github.com/maykinmedia/open-klant/pull/190
- [x] Open Notificaties https://github.com/open-zaak/open-notificaties/pull/163
- [ ] Open Zaak https://github.com/open-zaak/open-zaak/pull/1696
- [x] Objects https://github.com/maykinmedia/objects-api/pull/413
- [x] Objecttypes https://github.com/maykinmedia/objecttypes-api/pull/102
- [ ] Referentielijsten https://github.com/maykinmedia/referentielijsten/pull/17

**Comment by stevenbal** [comment]
@joeribekker is this just about environment variables, or also about Django settings?

**Comment by stevenbal** [comment]
I made a PR to OAf for this, I added a script that makes the matrix and included the matrix itself as well: https://github.com/maykinmedia/open-api-framework/pull/18 (formatted matrix: https://github.com/maykinmedia/open-api-framework/blob/feature/envvar-matrix/reports/envvar_matrix.csv)

**Comment by stevenbal** [comment]
I made a first pass through the differences and added the most obvious ones that need to be added/removed to this issue (some do require a bit more work, like adding setup_configuration)

**Comment by stevenbal** [comment]
PR in Open Klant: https://github.com/maykinmedia/open-klant/pull/187

**Comment by joeribekker** [comment]
> @joeribekker is this just about environment variables, or also about Django settings?

Well, in the end env settings but some might not be env settings yet everywhere :)

**Comment by stevenbal** [comment]
As we briefly talked about yesterday, after the current differences are fixed in the settings, it might be a good idea to look into creating a single base settings file (in OAf?) that can be imported in each project, to avoid duplication

**Comment by joeribekker** [comment]
Refinement: decided to add some "base-settings" that are used in all projects, in OAf and import/include it in each project.

**Comment by joeribekker** [comment]
Sjoerd mentions SUB PATH?

**Comment by stevenbal** [comment]
@joeribekker good call, I see that the envvar is loaded everywhere, but that Open Klant (and maybe other components) don't actually use the value of the envvar

**Comment by stevenbal** [comment]
I split the autodocumentation part off into a separate issue: https://github.com/open-zaak/open-zaak/issues/1649

**Comment by stevenbal** [comment]
PRs for Objects and Open Klant to use the generic settings from OAf are ready for review (and has already been merged for Objecttypes):
- https://github.com/maykinmedia/open-klant/pull/190
- https://github.com/maykinmedia/objects-api/pull/413

Doing this for Open Zaak and Open Notificaties is currently blocked because OAf isn't added there yet: https://github.com/maykinmedia/open-api-framework/issues/12

I'll mark this issue as blocked for now

**Comment by joeribekker** [comment]
Refinement: Open Notificaties and Open Zaak (draft) now have PRs so if they are merged this ticket can be unblocked.

**Comment by stevenbal** [comment]
The PRs in OZ and ON to add OAf are merged, so this is unblocked again
